### ISSUE TITLE
Assertion failed 'ssaNum != SsaConfig::RESERVED_SSA_NUM

### ISSUE THREAD

**Issue by kunalspathak** [dotnet/runtime#57916]
```c#
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.
//
// This file is auto-generated.
// Seed: -1
//
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;
public class TestClass14445 {
    public struct S1 {
        public int int32_1;
    }
    public struct S2 {
        public short int16_2;
    }
    public struct S3 {
    }
    public struct S4 {
        public short int16_1;
    }
    public struct S5 {
        public S1 s1_1;
    }
    static bool s_boolean_0 = true;
    static char s_char_2 = 'F';
    static S1 s_s1_14 = new S1();
    static S2 s_s2_15 = new S2();
    static S3 s_s3_16 = new S3();
    static S5 s_s5_18 = new S5();
    static int s_loopInvariant = 1;
    public int LeafMethod6() {
        unchecked {
            return 15 + 4;
        }
    }
    [MethodImpl(MethodImplOptions.NoInlining)]
    public S3 LeafMethod16() {
        unchecked {
            return s_s3_16;
        }
    }
    public short Method1(S4 p_s4_0) {
        unchecked {
            switch (s_char_2) {
                case 'E':
                    break;
                case '5':
                    break;
                case 'A':
                    break;
                default:
                    try {
                        if (s_boolean_0 = s_boolean_0 = true) {
                        } else {
                            if (s_boolean_0) {
                            } else {
                                try {
                                    int int32_21 = 15 + 4;
                                } catch (ArgumentException) {
                                } catch (MulticastNotSupportedException) {
                                } finally {
                                }
                            }
                            try {
                                s_s5_18.s1_1.int32_1 >>= s_s1_14.int32_1 &= 4;
                            } catch (OutOfMemoryException) {
                            } catch (ExecutionEngineException) {
                            } finally {
                            }
                        }
                        {
                            int __loopvar2 = s_loopInvariant;
                            for (; ; )
                            {
                                if (__loopvar2 >= 15 + 4) {
                                    break;
                                }
                                try {
                                    LeafMethod6();
                                } finally {
                                    LeafMethod16();
                                }
                            }
                        }
                    } catch (MethodAccessException) {
                    } catch (TypeUnloadedException) {
                    } finally {
                    }
                    break;
            }
            return s_s2_15.int16_2 *= 15 + 4;
        }
    }
    public void Method0() {
        unchecked {
            S4 s4_3 = new S4();
            s4_3.int16_1 = Method1(s4_3);
            return;
        }
    }
    public static void Main(string[] args) {
        TestClass14445 objTestClass14445 = new TestClass14445();
        objTestClass14445.Method0();
    }
}
/*
Got output diff:
--------- Baseline ---------  

Environment:



--------- Test ---------  

Environment:



Assert failure(PID 36284 [0x00008dbc], Thread: 35368 [0x8a28]): Assertion failed 'ssaNum != SsaConfig::RESERVED_SSA_NUM' in 'TestClass14445:Method1(S4):short:this' during 'Do value numbering' (IL size 163)

    File: D:\git\runtime\src\coreclr\jit\compiler.h Line: 352
    Image: D:\git\runtime\artifacts\tests\coreclr\windows.x86.Checked\tests\Core_Root\CoreRun.exe


*/

```

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Comment by kunalspathak** [comment]
The repro is on windows/x86

**Comment by jakobbotsch** [comment]
Probably fixed by #57274?

**Comment by kunalspathak** [comment]
Yep, it doesn't repro with latest main.
